pkg/controller/installation/core_controller.go

```diff
@@ -1699,6 +1726,52 @@ func (r *ReconcileInstallation) setDefaultsOnFelixConfiguration(install *operato
 			}
 		}
 	}
+
+	// If BPF is enabled, but not set on FelixConfiguration, do so here. Older versions of the operator
+	// used an environment variable to enable BPF, but we no longer do so. In order to prevent disruption
+	// when the environment variable is removed by the render code, make sure
+	// FelixConfiguration has the correct value set.
+	bpfEnabledOnDaemonSet, err := bpfEnabledOnDaemonSet(ds)
+	if err != nil {
+		reqLogger.Error(err, "An error occurred when querying the Daemonset resource")
+	} else if bpfEnabledOnDaemonSet && !bpfEnabledOnFelixConfig(fc) {
+		err = setBPFEnabled(fc, true)
+		if err != nil {
+			reqLogger.Error(err, "Unable to enable eBPF data plane")
+		} else {
+			updated = true
+		}
+	}
+
+	return updated
+}
+
+// setBPFUpdatesOnFelixConfiguration will take the passed in fc and update any BPF properties needed
+// based on the install config and the daemonset.
+func (r *ReconcileInstallation) setBPFUpdatesOnFelixConfiguration(install *operator.Installation, ds *appsv1.DaemonSet, fc *crdv1.FelixConfiguration, reqLogger logr.Logger) bool {
+	updated := false
+
+	bpfEnabledOnInstall := install.Spec.BPFEnabled()
+	if bpfEnabledOnInstall {
+		if (fc.Spec.BPFEnabled == nil || !(*fc.Spec.BPFEnabled)) && isRolloutComplete(ds) {
+			err := setBPFEnabled(fc, bpfEnabledOnInstall)
+			if err != nil {
+				reqLogger.Error(err, "Unable to enable eBPF data plane")
```

Comment:
Unfortunately, I think we need to propagate this error back up to `utils.PatchFelixConfiguration`. It means a slight tweak to the `patchFn` definition